engine/src/main/java/org/terasology/rendering/dag/DefaultStateChangeStorage.java

```diff
@@ -15,20 +15,19 @@
  */
 package org.terasology.rendering.dag;
 
-import java.util.List;
+import java.util.HashMap;
+import static org.lwjgl.opengl.GL11.GL_FILL;
 
 /**
- *
+ * TODO: Add javadocs
  */
-public class RenderPipelineExecutor {
-
-    private List<PipelineTask> taskList;
+class DefaultStateChangeStorage extends HashMap<Class<? extends StateChange>, StateChange> {
```

Comment:
Actually, I just noticed a problem in the code above. The goal is that defaultInstance is a singleton stored at class level. So, if it works you could do something like this:

`private static final defaultInstance = SetColorStateChange(hardcodedDefaultValue)`

Alternatively you could check in the constructor if defaultInstance is null and do the assignment only in that case. And there might be other ways. But the code above isn't quite right.